packages/mdc-menu-surface/foundation.js

```diff
@@ -178,16 +180,32 @@ class MDCMenuSurfaceFoundation extends MDCFoundation {
     this.anchorMargin_.left = typeof margin.left === 'number' ? margin.left : 0;
   }
 
+  /**
+   * Used to indicate if the menu-surface is hoisted to the body.
+   * @param {boolean} isHoisted
+   */
   setIsHoisted(isHoisted) {
-    this.close();
     this.hoistedElement_ = isHoisted;
   }
 
+  /**
+   * Used to set the menu-surface calculations based on a fixed position menu.
+   * @param {boolean} isFixedPosition
+   */
   setFixedPosition(isFixedPosition) {
-    this.close();
     this.isFixedPosition_ = isFixedPosition;
   }
 
+  /**
+   * Sets the menu-surface position on the page.
+   * @param {number} x
+   * @param {number} y
+   */
+  setAbsolutePosition(x, y) {
+    this.position_.x = isNaN(x) ? 0 : x;
```

Comment:
Change `isNaN(x)` to `!Number.isFinite(x)`.

From [`isNaN()` on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN):

> The `isNaN()` function determines whether a value is `NaN` or not. Note: coercion inside the isNaN function has **interesting** rules; you may alternatively want to use `Number.isNaN()`, as defined in ECMAScript 2015.

From [`Number.isFinite()` on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isFinite):

> In comparison to the global `isFinite()` function, `Number.isFinite()` doesn't forcibly convert the parameter to a number. This means only values of the type `number`, that are also finite, return `true`.

So it seems like `Number.isFinite()` is the closest to what you want: checking if the value is a usable x/y coordinate of type `number`.